app/src/main/java/com/github/se/eduverse/model/MediaCacheManager.kt

```diff
@@ -112,18 +155,43 @@ class MediaCacheManager(
     return file.delete()
   }
 
+  @OptIn(UnstableApi::class)
   fun cleanCache() {
     val currentTime = System.currentTimeMillis()
     val cacheDir = context.cacheDir
+
     cacheDir.listFiles()?.forEach { file ->
-      if (currentTime - file.lastModified() > CACHE_LIFETIME_MS) {
-        file.delete()
+      val timestamp = file.name.split("_").firstOrNull()?.toLongOrNull()
+      if (timestamp != null) {
+        val age = currentTime - timestamp
+        Log.d("CACHE_CLEAN", "File: ${file.name}, Age: $age ms")
+
+        if (age > CACHE_LIFETIME_MS) {
+          if (file.delete()) {
+            Log.d("CACHE_CLEAN", "Deleted expired file: ${file.name}")
+          } else {
+            Log.e("CACHE_CLEAN", "Failed to delete file: ${file.name}")
+          }
+        }
+      } else {
+        Log.e("CACHE_CLEAN", "Invalid file name, cannot extract timestamp: ${file.name}")
```

Comment:
Provides detailed logs for the cache cleaning process, including file age and deletion status.
Logs errors for invalid file names, helping identify problematic files during cleanup.
Attempts to parse the timestamp safely with toLongOrNull(), avoiding exceptions for invalid formats.
Deletes files exceeding a defined cache lifetime (CACHE_LIFETIME_MS), ensuring efficient storage management.